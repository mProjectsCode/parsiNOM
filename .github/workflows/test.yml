name: Test

on:
  push:
    branches: [master, main]
  pull_request:
    branches: [master, main]

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: oven-sh/setup-bun@v1
        with:
          bun-version: latest
      - working-directory: .
        run: bun install
      - working-directory: .
        run: bun run check
