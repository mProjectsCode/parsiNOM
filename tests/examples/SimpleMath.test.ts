import { describe, test, expect } from 'bun:test';
import { SimpleMathParser } from '../../examples/SimpleMath';

describe('simple math test', () => {
	const testCases: string[] = ['1 + 2', '1 + 2 + 3', '1 * 2 + 3', '1 + 2 * 3', '(1 + 2) * 3', '1 + -2', '1 + -2 ^ 2', '1 + -2 ^ 2!'];

	for (const testCase of testCases) {
		test(testCase, () => {
			const res = SimpleMathParser.tryParse(testCase);
			expect(res.success).toBe(true);
			expect(res.value).toMatchSnapshot();
		});
	}
});
