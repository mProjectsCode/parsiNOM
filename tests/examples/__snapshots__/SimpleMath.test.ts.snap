// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`simple math test 1 + 2 1`] = `
[
  "Add",
  [
    "Number",
    1,
  ],
  [
    "Number",
    2,
  ],
]
`;

exports[`simple math test 1 + 2 + 3 1`] = `
[
  "Add",
  [
    "Add",
    [
      "Number",
      1,
    ],
    [
      "Number",
      2,
    ],
  ],
  [
    "Number",
    3,
  ],
]
`;

exports[`simple math test 1 * 2 + 3 1`] = `
[
  "Add",
  [
    "Multiply",
    [
      "Number",
      1,
    ],
    [
      "Number",
      2,
    ],
  ],
  [
    "Number",
    3,
  ],
]
`;

exports[`simple math test 1 + 2 * 3 1`] = `
[
  "Add",
  [
    "Number",
    1,
  ],
  [
    "Multiply",
    [
      "Number",
      2,
    ],
    [
      "Number",
      3,
    ],
  ],
]
`;

exports[`simple math test (1 + 2) * 3 1`] = `
[
  "Multiply",
  [
    "Add",
    [
      "Number",
      1,
    ],
    [
      "Number",
      2,
    ],
  ],
  [
    "Number",
    3,
  ],
]
`;

exports[`simple math test 1 + -2 1`] = `
[
  "Add",
  [
    "Number",
    1,
  ],
  [
    "Negate",
    [
      "Number",
      2,
    ],
  ],
]
`;

exports[`simple math test 1 + -2 ^ 2 1`] = `
[
  "Add",
  [
    "Number",
    1,
  ],
  [
    "Exponentiate",
    [
      "Negate",
      [
        "Number",
        2,
      ],
    ],
    [
      "Number",
      2,
    ],
  ],
]
`;

exports[`simple math test 1 + -2 ^ 2! 1`] = `
[
  "Add",
  [
    "Number",
    1,
  ],
  [
    "Exponentiate",
    [
      "Negate",
      [
        "Number",
        2,
      ],
    ],
    [
      "Factorial",
      [
        "Number",
        2,
      ],
    ],
  ],
]
`;
